### PR TITLE
Add documents resource & data-source

### LINKED ISSUE #211: resource packetfabric_document
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

**Upload Documents**

- resource `packetfabric_document`
POST [/v2/documents](https://docs.packetfabric.com/api/v2/redoc/#tag/Documents/operation/document_upload)
⚠️There is no DELETE.

### PR DESCRIPTION
**Pull request by solababs**
## Description
Add documents resource

Closes #211  

## Checklist

- [x] tested locally
- [x] added/updated examples
- [x] added/updated docs

### CODE REVIEW

**Comment by rjouhann** [comment]
I tested both the new resource and data-source and get the following errors:

1. resource `packetfabric_document`

```
packetfabric_document.loa1: Creating...
╷
│ Error: Status: 400, body: {"message": "InvalidParam: missing \"document\" component of multipart upload", "request_id": "32fcebf7-077b-43d1-80b0-3f1eb412cf86"}
│ 
│   with packetfabric_document.loa1,
│   on test.tf line 282, in resource "packetfabric_document" "loa1":
│  282: resource "packetfabric_document" "loa1" {
│ 
```

2. data-source `packetfabric_document`

```
╷
│ Error: json: cannot unmarshal array into Go value of type packetfabric.Documents
│ 
│   with data.packetfabric_document.current,
│   on test.tf line 290, in data "packetfabric_document" "current":
│  290: data "packetfabric_document" "current" {
```

**Review by caitlinwheeless — APPROVED** [review]
Caitlin APPROVED! (nice links!) -- take 2

File: internal/packetfabric/client.go [hunk 1]
```diff
@@ -135,6 +150,65 @@ func (c *PFClient) sendRequest(uri, method string, payload interface{}, resp int
 	return res, nil
 }
 
+func (c *PFClient) sendMultipartRequest(uri, method, fileField, filePath string, payload map[string]string, resp interface{}) (interface{}, error) {
+	var req *http.Request
+	var err error
+	c.Ctx = context.Background()
+	formatedURL := fmt.Sprintf("%s%s", c.HostURL, uri)
```
Comment (medzin) [inline internal/packetfabric/client.go:157]: Consider using: https://pkg.go.dev/net/url#JoinPath

Comment (solababs) [reply internal/packetfabric/client.go]: Adding JoinPath fails for some reason, see https://github.com/PacketFabric/terraform-provider-packetfabric/pull/485/commits/80efad31650c13493e4ed5720ed0ad53fdacf33e, removing for now

**Review by rjouhann — APPROVED** [review]

**Pull request merged**